canonical-web-and-design/maas-ui
# Add segmented prop to Meter component.

## Done

- Added `segmented` prop to Meter component, which separates each discrete value in a meter with a 1px separator (whose colour can be customised with the `separatorColor` prop).
- Updated Meter labels and CPU header label in KVM list
- Updated `formatBytes` util to be able to convert to a specific unit, instead of just the nearest unit in which the value would be between 1 - 1023. This was needed to update the Meter labels so that you could do e.g. "0.1 of 10 GB assigned" instead of "100 MB of 10 GB assigned".
- Updated loading logic for KVMList cells that rely on host data. Now if the `pod` object does not have a `host` key, the cell value automatically goes to "Unknown" instead of waiting for all the controllers/machines to load first. 

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /MAAS/r/kvm and check that the meters match the [design](https://app.zeplin.io/project/5ed922c192bf80b9c37cb4fb/screen/5efa229d3f80d013734e9b21)
- Check that the table looks good on all screen sizes

## Fixes

Fixes canonical-web-and-design/MAAS-squad#1949

## Screenshot
![0 0 0 0_8400_MAAS_r_kvm (14)](https://user-images.githubusercontent.com/25733845/86999909-1ca6d580-c1f7-11ea-8296-97bcc8b84933.png)
